cetmix_tower_server/models/cx_tower_plan_line_action.py

```diff
@@ -60,19 +59,53 @@ def _compute_name(self):
 
             # Compose name based on values
             elif rec.condition and rec.action and rec.value_char:
-                action_string = action_selection_vals.get(rec.action)
-
-                # Add custom exit code if action presumes it
-                if rec.action == "ec":
-                    action_string = "{} {}".format(action_string, rec.custom_exit_code)
-                rec.name = " ".join(
-                    (
-                        _("If exit code"),
-                        rec.condition,
-                        rec.value_char,
-                        _("then"),
-                        action_string,
-                    )
+                rec.name = self._generate_action_name(
+                    rec.condition, rec.action, rec.value_char, rec.custom_exit_code
                 )
             else:
                 rec.name = _("Wrong action")
+
+    @api.model
+    def _generate_action_name(self, condition, action, value_char, custom_exit_code):
```

Comment:
Why do we need to make this a separate function?

Reply:
I'll remove this function